### PR TITLE
Fix TQDM Progress bar in TF 2.2.0

### PR DESCRIPTION
**Pull request by lminer**
TQDMProgressBar is currently broken in tensorflow 2.2.0 because of refactor of the way fitting works in tf.keras. This change should fix it so that it works as before.

### CODE REVIEW

**Comment by gabrieldemarmiesse** [comment]
Thanks for the fix! Could you add some tests to ensure that we won't intruduce any regressions in the future? You can take a look at how tests are made for layers or activations for example.

**Comment by bjtho08** [comment]
@lminer this fix disables reporting of validation metrics. Maybe you could try and look into this?

**Comment by gabrieldemarmiesse** [comment]
> this fix disables reporting of validation metrics. Maybe you could try and look into this?

Thanks a lot @bjtho08 for catching this before the merge. I added extensive testing, including a test for the issue about the validation metrics you're mentioning. See #1568 .

**Comment by gabrieldemarmiesse** [comment]
@googlebot I consent

**Comment by gabrieldemarmiesse** [comment]
It seems that not all the tests are passing with the proposed fix. Maybe somebody would like to step in to propose a new fix?

**Comment by seanpmorgan** [comment]
Did a small dig into this. The callback's `self.params` are not being set the same way in 2.1 and in 2.2.

2.2 self.params:
```
{'verbose': 0, 'epochs': 1, 'steps': 1}
```

2.1 self.params:
```
{'batch_size': 32, 'epochs': 1, 'steps': 1, 'samples': 5, 'verbose': 0, 'do_validation': False, 'metrics': ['loss', 'acc']}
```

Which is causing the validation metrics to not be reported:
https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/keras/callbacks.py#L160-L173

This affects more than the validation reporting since we leverage self.params in multiple places. E.g. the number of samples in a batch is no longer available so the batch progress always shows 1 sample per batch. 
`1/1  ██████████ ETA: 00:00s - loss: 0.4079 - acc: 0.2000`.

Using `self.model.metrics_names` won't be a sufficient fix. @lminer  do you have any more information regarding the changes to `keras.fit`?

**Comment by lminer** [comment]
@seanpmorgan Unfortunately I don't have any insight on this. Just noticed the breakage and offered an unsatisfactory fix :) Looking over the updated ProgBarCallback, it seems as if they don't rely on `metrics_names` anymore to decide what to log. Instead they just pop out size, num_steps, and batch and then log everything else regardless of the name: https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/keras/callbacks.py#L950

**Comment by bjtho08** [comment]
@lminer Can we not just do the same for TQDM progbar?

**Comment by gabrieldemarmiesse** [comment]
I'm looking into it, and I'll try to propose a pull request when I have a solution.

**Comment by gabrieldemarmiesse** [comment]
Superseded by #1595

**Comment by linyuanthocr** [comment]
Thanks for the update, colab uses 2.2.0.rc1 and has the problem as well. It totally works after the update!
